src/components/DayPickerRangeController.jsx

```diff
@@ -643,38 +651,55 @@ export default class DayPickerRangeController extends React.Component {
   }
 
   deleteModifier(updatedDays, day, modifier) {
-    const { numberOfMonths, enableOutsideDays } = this.props;
-    const { currentMonth, visibleDays } = this.state;
+    const { numberOfMonths: numberOfVisibleMonths, enableOutsideDays, orientation } = this.props;
+    const { currentMonth: firstVisibleMonth, visibleDays } = this.state;
+
+    let currentMonth = firstVisibleMonth;
+    let numberOfMonths = numberOfVisibleMonths;
+    if (orientation !== VERTICAL_SCROLLABLE) {
+      currentMonth = currentMonth.clone().subtract(1, 'month');
+      numberOfMonths += 2;
+    }
     if (!day || !isDayVisible(day, currentMonth, numberOfMonths, enableOutsideDays)) {
       return updatedDays;
     }
 
-    let monthIso = toISOMonthString(day);
-    let month = updatedDays[monthIso] || visibleDays[monthIso];
     const iso = toISODateString(day);
+
+    let updatedDaysAfterDeletion = { ...updatedDays };
     if (enableOutsideDays) {
```

Comment:
This addition to the `deleteModifier` method is very similar to what happens in `addModifier`. Namely, for outside days, sometimes a selected day might be in two `CalendarMonth`s and we need to remove the modifiers in both cases.